ydb/core/blobstorage/vdisk/balance/balancing_actor.cpp

```diff
@@ -1,29 +1,220 @@
+                        .Key=TLogoBlobID(It.GetCurKey().LogoBlobID(), partIdx),
+                        .Ingress=merger.Ingress,
+                        .PartData=*merger.Parts[partIdx - 1]
+                    });
+                    LOG_DEBUG_S(ctx, NKikimrServices::BS_VDISK_BALANCING,
+                            Ctx->VCtx->VDiskLogPrefix << "Send on main: " << SendOnMainParts.back().Key.ToString()
+                            << " " << SendOnMainParts.back().Ingress.ToString(&Ctx->GInfo->GetTopology(), Ctx->VCtx->ShortSelfVDisk, SendOnMainParts.back().Key));
+                }
+
+                for (ui8 partIdx: PartsToDelete(Ctx->GInfo->GetTopology(), Ctx->VCtx->ShortSelfVDisk, It.GetCurKey().LogoBlobID(), merger.Ingress)) {
+                    TryDeleteParts.push(TPartInfo{
+                        .Key=TLogoBlobID(It.GetCurKey().LogoBlobID(), partIdx),
+                        .Ingress=merger.Ingress,
+                    });
+                    LOG_DEBUG_S(ctx, NKikimrServices::BS_VDISK_BALANCING,
+                            Ctx->VCtx->VDiskLogPrefix << "Delete: " << TryDeleteParts.back().Key.ToString()
+                            << " " << TryDeleteParts.back().Ingress.ToString(&Ctx->GInfo->GetTopology(), Ctx->VCtx->ShortSelfVDisk, TryDeleteParts.back().Key));
+                }
+                merger.Clear();
+            }
+
+            Send(SelfId(), new NActors::TEvents::TEvCompleted());
+        }
+
+        void FinishCollecting(const TActorContext &ctx) {
+            LOG_DEBUG_S(ctx, NKikimrServices::BS_VDISK_BALANCING,
+                Ctx->VCtx->VDiskLogPrefix << ctx.Now().MilliSeconds() << " Bootstrap" << ": "
+                << "sendOnMainParts size = " << SendOnMainParts.size() << "; tryDeleteParts size = " << TryDeleteParts.size());
+
+            SenderId = ctx.Register(CreateSenderActor(SelfId(), std::move(SendOnMainParts), QueueActorMapPtr, Ctx));
+            ActiveActors.Insert(SenderId, __FILE__, __LINE__, ctx, NKikimrServices::BLOBSTORAGE);
+
+            DeleterId = ctx.Register(CreateDeleterActor(SelfId(), std::move(TryDeleteParts), QueueActorMapPtr, Ctx));
+            ActiveActors.Insert(DeleterId, __FILE__, __LINE__, ctx, NKikimrServices::BLOBSTORAGE);
+
             Become(&TThis::StateFunc);
+            Schedule(TDuration::Seconds(1), new NActors::TEvents::TEvWakeup());
         }
 
+        void StartBalancing(const TActorContext &ctx) {
+            LOG_DEBUG_S(ctx, NKikimrServices::BS_VDISK_BALANCING,
+                        Ctx->VCtx->VDiskLogPrefix << "Ask repl token");
+            if (!Send(MakeBlobStorageReplBrokerID(), new TEvQueryReplToken(Ctx->VDiskCfg->BaseInfo.PDiskId))) {
+                HandleReplToken(ctx);
+            }
+        }
+
+        void HandleReplToken(const TActorContext &ctx) {
+            LOG_DEBUG_S(ctx, NKikimrServices::BS_VDISK_BALANCING,
+                        Ctx->VCtx->VDiskLogPrefix << "Repl token acquired");
+            DoJobQuant(ctx);
```

Comment:
And, by the way, do we really want to interfere with actual replication?
Maybe we need to set up another queue (or priority queue) in the broker?
Otherwise this actor may speed down replication, which is much more relevant than blob balancing.